src/controllers/dynakube/version/image_hash.go

```diff
@@ -2,80 +2,72 @@ package version
-func GetImageVersion(ctx context.Context, imageName string, dockerConfig *dockerconfig.DockerConfig) (ImageVersion, error) {
-	transportImageName := fmt.Sprintf("docker://%s", imageName)
+func GetImageVersion(ctx context.Context, registryClient registry.ImageGetter, imageName string, dockerConfig *dockerconfig.DockerConfig) (registry.ImageVersion, error) {
+	keychain := dockerkeychain.NewDockerKeychain(dockerConfig.RegistryAuthPath, afero.NewOsFs())
+	transport := http.DefaultTransport.(*http.Transport).Clone()
 
-	imageReference, err := alltransports.ParseImageName(transportImageName)
+	ref, err := name.ParseReference(imageName)
 	if err != nil {
-		return ImageVersion{}, errors.WithStack(err)
+		return registry.ImageVersion{}, fmt.Errorf("parsing reference %q: %w", imageName, err)
 	}
 
-	systemContext := dockerconfig.MakeSystemContext(imageReference.DockerReference(), dockerConfig)
-
-	imageSource, err := imageReference.NewImageSource(context.TODO(), systemContext)
-	if err != nil {
-		return ImageVersion{}, errors.WithStack(err)
+	if dockerConfig.Dynakube.HasProxy() {
+		proxy, err := dockerConfig.Dynakube.Proxy(ctx, dockerConfig.ApiReader)
+		if err != nil {
+			return registry.ImageVersion{}, err
+		}
+		proxyUrl, err := url.Parse(proxy)
+		if err != nil {
+			log.Info("invalid proxy spec", "proxy", proxy)
+			return registry.ImageVersion{}, err
+		}
+
+		transport.Proxy = func(req *http.Request) (*url.URL, error) {
+			return proxyUrl, nil
+		}
 	}
-	defer closeImageSource(imageSource)
 
-	imageManifest, _, err := imageSource.GetManifest(context.TODO(), nil)
-	if err != nil {
-		return ImageVersion{}, errors.WithStack(err)
+	if dockerConfig.Dynakube.Spec.TrustedCAs != "" {
+		transport, err = addCertificates(transport, dockerConfig.Dynakube, dockerConfig.ApiReader)
+		if err != nil {
+			return registry.ImageVersion{}, fmt.Errorf("addCertificates(): %w", err)
+		}
 	}
 
-	digest, err := manifest.Digest(imageManifest)
-	if err != nil {
-		return ImageVersion{}, errors.WithStack(err)
-	}
+	return registryClient.GetImageVersion(ctx, ref, keychain, transport)
+}
 
-	sourceImage, err := image.FromUnparsedImage(context.TODO(), systemContext, image.UnparsedInstance(imageSource, nil))
+func addCertificates(transport *http.Transport, dynakube *dynakube.DynaKube, apiReader client.Reader) (*http.Transport, error) {
+	trustedCAs, err := dynakube.TrustedCAs(context.TODO(), apiReader)
 	if err != nil {
-		return ImageVersion{}, errors.WithStack(err)
+		return transport, err
 	}
 
-	inspectedImage, err := sourceImage.Inspect(context.TODO())
-	if err != nil {
-		return ImageVersion{}, errors.WithStack(err)
-	} else if inspectedImage == nil {
-		return ImageVersion{}, errors.Errorf("could not inspect image: '%s'", transportImageName)
+	rootCAs := x509.NewCertPool()
+	if ok := rootCAs.AppendCertsFromPEM(trustedCAs); !ok {
+		log.Info("failed to append custom certs!")
 	}
+	transport.TLSClientConfig.RootCAs = rootCAs
```

Comment:
I suppose that "if nil" [condition](https://github.com/Dynatrace/dynatrace-operator/blob/main/src/dtclient/client.go#L204) should be used here

Reply:
Fixed [here](https://github.com/Dynatrace/dynatrace-operator/pull/1998/commits/ad2c7d1a66a1b40faf3835303b4ac9e5d3703718)

Reply:
yeah agree with @aorcholski ☝️  good point